resources/aws_ec2.go

```diff
@@ -486,6 +489,68 @@ func (obj *AwsEc2Res) longpollWatch() error {
 	}
 }
 
+// snsWatch uses amazon cloudwatch events and simple notification service to
+// detect ec2 instance state changes.
+func (obj *AwsEc2Res) snsWatch() error {
+	send := false
+	var exit *error
+	defer obj.wg.Wait()
+	defer close(obj.closeChan)
+	// set up the sns endpoint
+	snsEndpoint := obj.snsServer()
+	// shutdown the sns endpoint when we're done
+	obj.wg.Add(1)
+	go func() {
+		defer obj.wg.Done()
+		select {
+		case <-obj.closeChan:
+			if err := snsEndpoint.Shutdown(nil); err != nil {
```

Comment:
Seeing this misc `nil` here was a red flag. I hope you're reading the API docs instead of doing copy+pasta from stackoverflow ;) -- https://golang.org/pkg/net/http/#Server.Shutdown It says: `Make sure the program doesn't exit and waits instead for Shutdown to return.`

So looks like we probably want to add a context with a timeout here instead of `nil` and do the safe thing and wait.

Reply:
Cool. Will do. BTW, I got this from the somewhat cryptic HTTP docs, not stack :). Was aware of the context, but thought it was unnecessary (I was thinking of WithCancel, not timeout.) I'll add it.